### PR TITLE
Int 341 crowdstrike vulns

### PR DESCRIPTION
**Pull request by VDubber**
NOTE: We still do not have access to the spotlight API. This should only be merged if approved by @ninetreats .

- Create iterateVulns method in client
- Add Vulnerability type
- Create converter for vulnerabilities
- Add test for fetch/collecting vulns
- Update docs
- Update yarn.lock
- Added stepStartState to enable/disable vuln step via accountId. See [here](https://github.com/JupiterOne/integration-crowdstrike/pull/61) for more

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,34 +1,39 @@
...
-};
+    /**
+     * At the time of converting from the managed SDK to the open-source SDK, the implemented
+     * `invocationValidator` is an empty function.
+     *
+     * TODO implement validateInvocation()
+     */
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    validateInvocation: () => {},
+    getStepStartStates,
```
Comment (VDubber) [inline src/index.ts:31]: Add this line: getStepStartStates

File: src/getStepStartStates.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ */
+export default function getStepStartStates(
+  executionContext: IntegrationExecutionContext<CrowdStrikeIntegrationInstanceConfig>,
+): StepStartStates {
+  return {
+    [getAccountStep.id]: { disabled: false },
+    [fetchDevicesStep.id]: { disabled: false },
+    [fetchDevicePolicyRelationshipsStep.id]: { disabled: false },
+    [fetchPreventionPoliciesStep.id]: { disabled: false },
+    [fetchVulnerabilitiesStep.id]: {
```
Comment (VDubber) [inline src/getStepStartStates.ts:25]: See [here](https://github.com/JupiterOne/integration-crowdstrike/pull/61) for context

File: src/steps/fetchVulnerabilities.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    query: {
+      filter: `created_timestamp:>'${createdTimestampFilter}'`,
+    },
+    callBack: async (vulns) => {
+      logger.info(
+        { vulnerabilityCount: vulns.length, createdTimestampFilter },
+        'Creating vulnerability entities and relationships...',
+      );
+      for (const vulnerability of vulns) {
+        const vulnerabilityEntity = await jobState.addEntity(
```
Comment (VDubber) [inline src/steps/fetchVulnerabilities.ts:40]: Add logging of the raw vuln for model understanding

File: src/jupiterone/converters.ts [hunk 4]
```diff
@@ -131,6 +132,32 @@ export function createPreventionPolicyEntity(source: PreventionPolicy): Entity {
...
 }
 
+export function createVulnerabilityEntity(source: Vulnerability) {
+  return createIntegrationEntity({
+    entityData: {
+      source,
+      assign: {
+        _class: Entities.VULNERABILITY._class,
+        _type: Entities.VULNERABILITY._type,
+        _key: source.id,
```
Comment (adam-in-ict) [inline src/jupiterone/converters.ts:142]: Just confirming this is a UUID and guaranteed to be universally unique and not just an ID # that may not be unique across everything.

**Review by adam-in-ict — DISMISSED** [review]

File: src/jupiterone/converters.ts (hunk 4 above)
Comment (VDubber) [reply src/jupiterone/converters.ts:142]: I'll double check.

Comment (VDubber) [reply src/jupiterone/converters.ts:142]: They are a UUID according to the docs. I'll double check once we get real data.

File: src/steps/fetchVulnerabilities.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+          createVulnerabilityEntity(vulnerability),
+        );
+
+        const sensor = await jobState.findEntity(vulnerability.aid);
```
[the remark this answers is not in the document]
Comment (VDubber) [reply src/steps/fetchVulnerabilities.ts:51]: Added a TODO statement to remember this 👍

**Review by adam-in-ict — APPROVED** [review]

**Pull request merged**